xbmc/platform/android/drm/MediaDrmCryptoSession.cpp

```diff
@@ -199,18 +206,29 @@ bool CMediaDrmCryptoSession::Verify(const Buffer &macKeyId, const Buffer &messag
 //Private stuff
 bool CMediaDrmCryptoSession::OpenSession()
 {
+  bool provisioned = false;
+TRYAGAIN:
   m_sessionId = new CharVecBuffer(m_mediaDrm->openSession());
   if (xbmc_jnienv()->ExceptionCheck())
   {
-    delete m_sessionId, m_sessionId = nullptr;
     xbmc_jnienv()->ExceptionClear();
-    return false;;
+    if (provisioned || !ProvisionRequest())
+    {
+      delete m_sessionId, m_sessionId = nullptr;
+      return false;
+    }
+    else
+    {
+      provisioned = true;
+      goto TRYAGAIN;
```

Comment:
You leak a CharVecBuffer instance for every retry now, right?

Reply:
Thanks for the hint, I'll change but its not critical because provisioning is done exactly once per device after factory reset.

Reply:
Yeah np, just catched it by accident ;)